upgrade-tests/src/test/java/io/zeebe/test/UpgradeTest.java

```diff
@@ -137,73 +203,118 @@ private static BpmnModelInstance jobWorkflow() {
-    private Function<ContainerStateRule, Long> before;
-    private BiConsumer<ContainerStateRule, Long> after;
+  private static long messageSubscription(final ContainerStateRule state) {
+    TestUtil.waitUntil(() -> state.hasLogContaining("WORKFLOW_INSTANCE_SUBSCRIPTION", "OPENED"));
+    return -1L;
+  }
 
-    TestCase createInstance(final BpmnModelInstance model) {
-      this.createInstance =
-          client -> {
-            client.newDeployCommand().addWorkflowModel(model, PROCESS_ID + ".bpmn").send().join();
-            client
-                .newCreateInstanceCommand()
-                .bpmnProcessId(PROCESS_ID)
-                .latestVersion()
-                .send()
-                .join();
-          };
-      return this;
-    }
+  private static void publishMessage(
+      final ContainerStateRule state, final long wfInstanceKey, final long key) {
+    state
+        .client()
+        .newPublishMessageCommand()
+        .messageName(MSG)
+        .correlationKey("123")
+        .timeToLive(Duration.ofMinutes(5))
+        .send()
+        .join();
 
-    /**
-     * Should make zeebe write records and write to state of the feature being tested (e.g., jobs,
-     * messages). The workflow should be left in a waiting state so Zeebe can be restarted and
-     * execution can be continued after. Takes the container rule as input and outputs a long which
-     * can be used after the upgrade to continue the execution.
-     */
-    TestCase beforeUpgrade(final Function<ContainerStateRule, Long> func) {
-      this.before = func;
-      return this;
-    }
-    /**
-     * Should continue the instance after the upgrade in a way that will complete the workflow.
-     * Takes the container rule and a long (e.g., a key) as input.
-     */
-    TestCase afterUpgrade(final BiConsumer<ContainerStateRule, Long> func) {
-      this.after = func;
-      return this;
-    }
+    TestUtil.waitUntil(() -> state.hasMessageInState(MSG, "PUBLISHED"));
+  }
 
-    public Consumer<ZeebeClient> createInstance() {
-      return createInstance;
-    }
+  private static BpmnModelInstance msgStartWorkflow() {
+    return Bpmn.createExecutableProcess(PROCESS_ID)
+        .startEvent()
+        .message(b -> b.zeebeCorrelationKey("key").name(MSG))
+        .endEvent()
+        .done();
+  }
 
-    public Function<ContainerStateRule, Long> before() {
-      return before;
-    }
+  private static long startMsgSubscription(final ContainerStateRule state) {
+    TestUtil.waitUntil(() -> state.hasLogContaining("MESSAGE_START_EVENT_SUBSCRIPTION", "OPENED"));
+    return -1L;
+  }
 
-    public BiConsumer<ContainerStateRule, Long> after() {
-      return after;
-    }
+  private static BpmnModelInstance timerWorkflow() {
+    return Bpmn.createExecutableProcess(PROCESS_ID)
+        .startEvent()
+        .timerWithCycle("R/PT1S")
+        .endEvent()
+        .done();
+  }
+
+  private static long timerCreated(final ContainerStateRule state) {
```

Comment:
Trying to aline the names.

```suggestion
  private static long awaitTimerCreated(final ContainerStateRule state) {
```